src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseTestCaseService.java

```diff
@@ -155,6 +155,9 @@ public boolean generateTestCasesFromFeedbacks(List<Feedback> feedbacks, Programm
         testCasesToSave.addAll(newTestCases);
         testCasesToSave.addAll(testCasesWithUpdatedActivation);
 
+        // Ensure no duplicate TestCase is present
```

Comment:
```suggestion
        // Ensure no duplicate TestCase is present: TestCases have to have a unique name per exercise.
        // Just using the uniqueness property of the set is not enough, as the equals/hash functions
        // consider more attributes of the TestCase rather than only the testName.
```
I think a bit more information why just using a set is not enough is helpful when someone else looks at the code in future.